backend/main.py

```
@@ -206,6 +212,83 @@ def get_mock_price_data(self, symbol: str) -> Dict[str, Any]:
+        logger.info(f"AlphaVantage: fetched {symbol} price {price}")
+        return result
+
+    def _get_stock_price_yfinance(self, symbol: str) -> Dict[str, Any]:
+        """Fetch latest price using yfinance as a live source (fallback)"""
+        # Note: yfinance may perform multiple network calls internally; wrap in try/except
+        ticker = yf.Ticker(symbol)
+        info = ticker.info
+        hist = ticker.history(period="1d")
+
+        if hist.empty:
+            raise ValueError(f"No data available for {symbol} via yfinance")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)